components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementAdminService.java

```diff
@@ -108,16 +124,23 @@ public void updateApplication(ServiceProvider serviceProvider)
             throws IdentityApplicationManagementException {
 
         // check whether use is authorized to update the application.
-        if (!ApplicationConstants.LOCAL_SP.equals(serviceProvider.getApplicationName()) &&
-                !ApplicationMgtUtil.isUserAuthorized(serviceProvider.getApplicationName(), getUsername(),
-                        serviceProvider.getApplicationID())) {
-            log.warn("Illegal Access! User " + CarbonContext.getThreadLocalCarbonContext().getUsername() +
-                    " does not have access to the application " +
-                    serviceProvider.getApplicationName());
-            throw new IdentityApplicationManagementException("User not authorized");
+        try {
+            if (!ApplicationConstants.LOCAL_SP.equals(serviceProvider.getApplicationName()) &&
+                    !ApplicationMgtUtil.isUserAuthorized(serviceProvider.getApplicationName(), getUsername(),
+                            serviceProvider.getApplicationID())) {
+                log.warn("Illegal Access! User " + CarbonContext.getThreadLocalCarbonContext().getUsername() +
+                        " does not have access to the application " +
+                        serviceProvider.getApplicationName());
+                throw new IdentityApplicationManagementException("User not authorized");
+            }
+            applicationMgtService = ApplicationManagementService.getInstance();
+            applicationMgtService.updateApplication(serviceProvider, getTenantDomain(), getUsername());
+        }catch (IdentityApplicationManagementException idpException){
+
+            log.error("Error while updating application by the service provider :" +serviceProvider  , idpException);
+            throw idpException;
+
```

Comment:
Please follow the same pattern for blank lines. Since your previous catch blocks are without any blank lines, please keep this and following occurrences without new lines.